### DISCUSSION TITLE
Allow issues to belong to a project and be "linked" to multiple repositories

### DISCUSSION THREAD

**Discussion by francos** [community/community#6433] [Projects and Issues]
In our organisation, we have multiple internal library repositories that are used from multiple app repositories. This allows us to share a lot of code and not have to write it multiple times, but it brings a problem when trying to deal with issues in Github as they belong to one repository only. Because of this limitation, we decided to move to another issue tracker that allows us to create an issue that can be referenced from the commits of multiple repositories.

It'd be great if this feature can be added to the new version of Github Projects/Issues as it'd allow teams like ours to create a single issue that belongs to the project and can be easily "linked" in the commits of multiple repositories :)

**Comment by hassannteifeh** [comment]
> In our organisation, we have multiple internal library repositories that are used from multiple app repositories. This allows us to share a lot of code and not have to write it multiple times, but it brings a problem when trying to deal with issues in Github as they belong to one repository only. Because of this limitation, we decided to move to another issue tracker that allows us to create an issue that can be referenced from the commits of multiple repositories.
> 
> It'd be great if this feature can be added to the new version of Github Projects/Issues as it'd allow teams like ours to create a single issue that belongs to the project and can be easily "linked" in the commits of multiple repositories :)

Yeah it's a bummer that even the new version still does not provide this very simple yet crucial functionality. For now I'll be using one repository as the single place to create issues across all the repos and will be using Github issues Beta for its boards/views.

**Reply by francos** [reply]
@hugonteifeh how do you reference your issues from the project commits without it taking too much of the commit space?

**Comment by ShaneZhengNZ** [comment]
Very true. It is a little disappointing that this feature is still missing. I don't want to use the third-party tool for project management since it makes sense to have the issues, tickets, and everything else that has to do with code live with the repositories in GitHub. But missing this important feature, I have to use something else for now.

**Comment by beyjor** [comment]
```
It'd be great if this feature can be added to the new version of Github Projects/Issues as it'd allow teams like ours to create a single issue that belongs to the project and can be easily "linked" in the commits of multiple repositories :)
```

Ideally it would also allow the 'Break issues into actionable tasks' capability to break a task list out into issues for multiple repos while keeping the parent feature at the project level and maintaining the tracking tree. 
![image](https://user-images.githubusercontent.com/44268517/139907604-76c326a1-e925-4866-b0d3-43a973dba5b6.png)

**Reply by nahtnam** [reply]
Yeah this is what is pushing me to linear

**Reply by linus-guestrix** [reply]
This is possible to do now with the "Tasklist" feature. It is currently in beta, but I got access only a day after applying. You should check it out, it looks really promising. What you can do is create an issue in one repo and the create a tasklist in that issue with issues in other repos. You can also track progress based on the status of the linked issues.

**Reply by BlockchainSpaceDev** [reply]
> linear

I'm curious, did u switch to it at the end? How did it go?

**Comment by timbze** [comment]
Very good idea. I would add, it would be great that each project could have it's own setting.

For example, I have 2 projects that I want to always reference issues in my "Issues" project. But these 2 projects still have pull requests, so the PR # would conflict with the issue numbers in my "Issues" project.

My suggestion:
1. Set the 2 repositories to have their secondary repo be my "Issues" repo
2. Reference "Issues" repo by I#[100], or some other letter preceding the #

Also, my 2 repos already have issues feature disabled, possibly only allow this when issues is disabled.
Or, make the letter preceding the # configurable for as many other repos as you want. E.g. If I want to reference "Issues" repo, use I#[100], if I want to reference planning repo, P#[88], etc

**Comment by LeeU1911** [comment]
I'm surprised by low number of comments on this feedback. Maybe almost everyone is one another project/issue trackers and only use Github Issues for bugs/feature requests.
Coming from JIRA, I could only use Github Issues for things that are mostly independent

**Comment by tvo318** [comment]
Even if the fields supported markdown like in the actual GitHub tickets, you can link to another repo with a shorter custom text, would be better than nothing!

**Comment by matthias-jauernig** [comment]
This is really a *huge* bummer to use Github issues. Currently it's just a simple issue tracker, but no project management tool, where you can break down things and define epics, stories and tasks. Already just a simple story/task separation would be great.

**Reply by BlockchainSpaceDev** [reply]
Call it "project management tool" is literally a blasphemy. It's full of limitations and the one mentioned here is just a piece of the puzzle:
- headaches to make milestones cross-repo
- headeaches dealing with templates multi-repo
- only parent-child relationship whilst competitors since ages show also dependency-relations and so on
but the list is long and annoying

Evaluating to switch to Linear as well.

**Comment by couling** [comment]
I think this feature request is receiving less love because for people to ask for a "feature" they are usually already able to use your software.

This limitation of github issues is so crippling that many of us can't even think about using github issues before it's implemented.

**Comment by beonde** [comment]
Upvote on this for sure! We are currently having to duplicate an issue across 2 to 3 repos to make sure we track it in our various apps that are affected by a change or a bug fix. Of course, that is a lot of extra work and we may miss a detail in a spec or feedback.

@github-staff Please please. Give this request some lovin.

**Comment by achimnol** [comment]
Project-level issue could be an alternative to project-level/cross-repo PR as suggested in #13733.

**Comment by PaulJNewell77** [comment]
It seems they are **so** close to having this feature. With the new project capability, they have the concept of 'draft' issues which exist only in the project (not in a repo). All they need to do is add the task-list feature to this (like they have for repo issues) - oh, and not call them draft!

**Comment by equetzal** [comment]
They really need to focus into this! :(

**Comment by adrianrudnik** [comment]
Second time I try to get into github projects, but how do teams actually use this to get anything organized?

Simple repo scenario: API and frontend, not a mono-repo. Now how am I supposed to create and organize a simple password-reset flow that requires actions and stories within two repos in parallel to get this shipped? I open up a repo-unbound project. I add draft tasks. From this point on I'm forced to use repo bound things: Issues, milestones, PRs and so on.

- I can't use the checkbox-to-issue-feature that is available in simple repo issues.
- There is no simple cross-referencing different repos, no easy linking issues, only manually copy/paste the other issue towards the list within the project.
- Checking of an item in the project comment (linked to an issue) does not close the issue, what is it for then?

it just does not feel like anything useful. Either a project has to be equal to an epic, or this is just a huge manual copy/paste scratchboard with no real influence on anything. Actually abusing projects as epic feels like the way it is intended, but somehow it also feels just wrong.

**Comment by Samreay** [comment]
It's crazy how this fundamental features still being lacking cripples GitHub Projects as being viable for any project or organisation that doesn't have a _single monorepo_.  We're still stuck with JIRA because of this, but I'm curious if anyone has found any decent work arounds outside of just using third party tools instead?

**Reply by francos** [reply]
Hey @Samreay, I ended up with a less-than-ideal solution: 

* I created a repository called "issues" which I only use to create issues for all the projects (there is no code in this repo). 
* Then I link PRs to those issues by using the notation `<org>/issues/<issue-number>` in the title and description of the PR.

This made it quite inconvenient for us to link commits to issues (because of the long notation which can easily have typos) so we ended up deciding to use "squash merge" from PRs which has worked pretty well and kept the commits on the `main` branch really clean.

This solution has its drawbacks though and it'd be much better if a native solution was provided by GitHub of course, but maybe you can give it a go and see if it works for you.

**Reply by karthiksivanadiyan** [reply]
You're right, really don't understand why this is especially in a world of micro-services where each service typically has its own repository.

**Reply by florianmonfort** [reply]
Indeed, I really wish GitHub would provide an answer to this. It's annoying every day especially when dealing with Epics haha.

**Comment by SteveBraich** [comment]
Upvoting this feature request.

**Comment by jordandakota** [comment]
My primary use case is being able to land a new issue at the project level and then assign it down to specific repos. This seems to work with draft issues, where we can add status, lists, assign, and then issue it down in the repo. However, it would also be super useful to have an issue IN the project so it could be referenced by multiple repos.

**Comment by VaZark** [comment]
Ideally, the issues would have a separate field `Project Item` so that the issues are local to the repository while the project has an overview of the task associated with it. 

Given that it could be a breaking change, it'd be great if we have `Epic` have tasks and let tasks be a one-to-one map to an issue

**Comment by florianmonfort** [comment]
Upvoting this. Task lists are a step in the right direction, but still not sufficient. We are still required to create an initial issue in a repo, whereas that main "Epic" issue is actually not really about a specific repository. It's just not logical.

**Comment by LeslieOA** [comment]
Sorry to be a "+1" thread human, but has there been any consideration of implementing this?
ZenHub seems overkill for something that could be relatively "simple" to implement. 
I wouldn't expect an issue to truly be shared between repositories, but it would be great to provide a way of showing an issue has a secondary, "symbolic link" to a repository in GitHub Project views.

**Comment by aprilmintacpineda** [comment]
The lack of this feature left me needing to create multiple tickets for one task, e.g., `[FE] Login` and `[BE] login` because while I'm the only one working on my project -- I didn't go with a monorepo setup. Maybe this is really the way they intended for us to use this -- create separate ticket for separate domain even though it's for the same concern.

EDIT: I just got one problem -- and it's that once you converted a draft to an issue and selected the corresponding repository where it would belong, you can no longer edit it. So if you make a mistake and selected the wrong repo, you will have to close that issue and create a new entry on the project and then copy-paste everything from the incorrect and then create an issue to the correct repository. I think being able to attach a ticket to multiple repositories will be helpful for this problem.

**Reply by cjwhitsitt** [reply]
You can transfer issues between repositories, with some caveats: https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository

**Comment by regicsolutions** [comment]
Could agree more, this is one thing Gitlab did right. Projects are a concept across all our tooling except GitHub where you can group objects, Organization doesn’t make much sense given its name alone, there is usually more than one project going on in each Organization I would think right?

**Comment by denzel-farmer** [comment]
Would also find this useful, it seems like GitHub wants "Projects" to be just a fancy way to view issues across a bunch of repositories, whereas most other project management software (GitLab, JIRA) view them (or their equivalent to projects) as "owning" the issues within their repositories. In my mind the second method seems much more powerful.

**Comment by garcondetroie** [comment]
It seems like they've addressed this. I can now link task from different repo using `myorgname/repo#123` and it will create a link to the corresponding task. So in the backend, referencing a front end issue I would write `myorgname/frontend#123`. I know it's not directly linked to multiple commits over multiple repositories, but it's the best way I found.

**Reply by florianmonfort** [reply]
@garcondetroie we've resorted to doing this, but we don't like it. It creates added complexity for our staff to maintain a million issues when we like to have things centralized into one.

**Reply by iprasla** [reply]
I agree. The tasks not necessarily accomplish that. There should be a way to link issues or PRs across difference repositories.

**Comment by mistvan** [comment]
looks like this product is there to solve our problems https://codetree.com/

**Reply by florianmonfort** [reply]
@mistvan no thanks, I don't want to have to subscribe to a separate product to be able to do something this basic. We already give plenty of money to GitHub.

**Comment by Murodxojayev1ch** [comment]
Response #1

**Comment by gz-jgale** [comment]
We have an org with many repos. Since we work with multiple 3rd parties, we have varying access to each repo. 

It is often not clear to our testers which repo we need to create an issue nor do our repo owners want issues created in the wrong repos. Because of this complexity we have been using a single repo the testers and a few managers have access to as a place [to create issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue) that are then reviewed and [transferred to different repos](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) where the right group of people will have access to work on it. 

🫠Our biggest challenge in using this approach is anything added in the original issue post (uploaded in-comment images, videos or files) DO NOT TRANSFER when we transfer the issue to another repo. 

🤬The OG issue repo has very limited access because we do not want all our 3rd party vendors we work with to see each others content. When we transfer an issue to a repo they do have access to, they can not see anything that was part of the issue before it was transferred forcing duplication of images or falling back to spreading issue content and discussion across MS teams, slack and email making it harder to follow. 

💡FEATURE REQUEST: When transferring (or linking) an issue to another repo, `all embedded content (images, videos, files etc) uploaded as part of the issue creation or comments on the issue, needs an issue content permission update to allow users of the new destination/linked repo to at least see the content`. 
* It could be done with or without a confirmation dialogue. Org disk space may fit into this if we are talking about making it work for cross-org transfers, maybe an approval process for intake would help?
* @j0siepy, this would effect content from forms and templates if we want allow issue managers to mark things as `non-transferable` or not.

**Comment by jayleaton** [comment]
Bump we are considering swapping to Github projects but this is a must have feature given we use many micro services its not practical to create new issues in every repo that needs changes for a single feature.

**Comment by vinniefalco** [comment]
I'm having the same problem. A Django website repository, and a separate repository for documentation and user interface templates.

**Comment by mibressler** [comment]
Has anybody figured out to do this? (e.g. with a custom linear Integration/Plugin)

**Comment by paguileradeta** [comment]
2025, still not implemented :(

**Comment by kungfuchicken** [comment]
+1

**Comment by m-tse** [comment]
I so wanted to use something simple like GitHub issues to track tasks. But this is a basic ask that GitHub just doesn't seem to care about :/

**Comment by EnriqueTMT** [comment]
+1 Commenting to show interest.

**Comment by KMastalerz** [comment]
Yup, this is true id like to be able to link issue to multiple repositories from my perspective this is one crucial implementation to being all we need.

Let me actually provide some "why" explanation:

We have multiple projects that are not separate, single services, receiver and multiple products around them: mobile, web and desktop.

When client raises a bug its more often than not change in both services and respective client. When we commit a change and it goes to QA backlog it should be one validation task, not two. 

I could create a project (BugTracker) for each product, and then link issue to it, but it would still be in one repository. Id like to see it in history of both. To be sure why change happened.

Although I see thats probably a lot of rework for entire architecture, but its a link table between repositories and issue we would all greatly appreciate:)
Look up

**Comment by tomkli-symphony** [comment]
I miss this feature so much, need to track few issues in different repositories with one parent item!

**Comment by wfzelle** [comment]
+1

**Comment by alexandre-bonnefond** [comment]
+1

**Comment by HeldtA** [comment]
+1

**Comment by BlockchainSpaceDev** [comment]
+1
A feature usually it's composed by at least a FE and BE implementation, without this implementation it means that we have to create duplicates of everything, "epics", user stories and so on.
That's dumb.

**Comment by olonyx** [comment]
This functionality would significantly enhance cross-repository project management, particularly for organizations with complex, interdependent codebases. It would reduce duplication, and align well with modern DevOps practices. Strongly support this.

**Comment by vonheim-sudo** [comment]
+1

**Comment by maoudia-via** [comment]
+1

**Comment by TheLaurenBarger** [comment]
+1

**Comment by vultpham** [comment]
+1

**Comment by alapollon** [comment]
Send this up!

**Comment by ncarvajalc** [comment]
+1

**Comment by rosjat** [comment]
+1

**Comment by J0eppp** [comment]
+1

**Comment by emil-st** [comment]
+1

**Comment by samuel27m** [comment]
+1

**Comment by amglt** [comment]
+1

**Comment by grabear** [comment]
+1

**Comment by beingminimal** [comment]
@mariorod @ashtom 
Please allocate few team members on making project management better also.
This message is a just gentle reminder and humble request also. Thanks

**Comment by dsfrederic-cgk** [comment]
+1

**Comment by ehnsio** [comment]
+1

**Comment by linus-guestrix** [comment]
Please let's get this done, it would help us a lot since we're doing all our product management in GitHub Projects.

**Comment by aaron-gar** [comment]
+1

**Comment by MarijnRS** [comment]
+1

**Comment by iqthomas** [comment]
+1

**Comment by franpek** [comment]
+1

**Comment by bms63** [comment]
+1

Came here looking to link three repos together in different meta-repos to one project board...and was left sad

**Comment by Bradley-McCallion** [comment]
Does anybody know if this is even in the roadmap?

**Comment by kprotsko** [comment]
Its very difficult to work with the way it is and easy for the issue to end up in the wrong repo that projects randomly chose that I had access to.      I was ready to drive all new teams into the projects  if it was not for this very large feature gap that is making me reevaluate going to a different solution.

**Comment by clausmichele** [comment]
I would be interested too!

**Comment by gbschenkel** [comment]
Sad this is still an Issue today, and Microsoft keeps pushing Azure Boards agenda into my employer to handle some basic stuffs, but can't auto link an board task with the issue making me create both, and manual link them. What decadence has become GitHub, let's see if they will push again the idea of billing the self hosted runners.

**Comment by UnknownHawkins** [comment]
Thanks for sharing this idea. We also face the same problem in our project. We use many repositories for different apps and some common internal libraries. This helps us reuse code, but it becomes difficult when we want to track issues.

Most of the time, one issue is related to more than one repository, but GitHub allows an issue to belong to only one repository. Because of this, we sometimes have to create multiple issues for the same problem, which is confusing.

If GitHub allows creating issues at the project level and linking them with commits from multiple repositories, it would be very helpful. It would make issue tracking easier and save time for teams like ours.

Hope this feature can be added in future updates.

**Comment by linde-ibis** [comment]
Is this ever going to happen? It would make our processes so much easier and so much confusion would be avoided!

**Comment by amrohendawi** [comment]
I am able to link PRs from different repositories within the same project using gh graphql CLI commands at the moment.
this is really a terrible way of working but it works for now.

```sh
#!/bin/bash

#############################################
# Link a Pull Request to an Issue (Cross-Repo)
#############################################

# CONFIGURE THESE VARIABLES
PR_OWNER=""                                 # GitHub organization/owner name
PR_REPO=""                                    # Repository containing the PR/branch
PR_NUMBER=""

ISSUE_OWNER=""                                 # GitHub organization/owner name
ISSUE_REPO=""                                    # Repository containing the issue to link the PR to
ISSUE_NUMBER=""

# Closing keyword (Fixes, Closes, Resolves, or "Relates to" for non-closing link)
KEYWORD="Fixes"

#############################################
# Script Logic (No changes needed below)
#############################################

echo "📎 Linking PR #${PR_NUMBER} (${PR_OWNER}/${PR_REPO}) to Issue #${ISSUE_NUMBER} (${ISSUE_OWNER}/${ISSUE_REPO})..."

# Get PR node ID and current body
RESPONSE=$(gh api graphql -f query="
query {
  repository(owner: \"${PR_OWNER}\", name: \"${PR_REPO}\") {
    pullRequest(number: ${PR_NUMBER}) {
      id
      body
    }
  }
}")

PR_ID=$(echo "$RESPONSE" | jq -r '.data.repository.pullRequest.id')
CURRENT_BODY=$(echo "$RESPONSE" | jq -r '.data.repository.pullRequest.body')

# Build issue reference
if [ "$PR_OWNER" = "$ISSUE_OWNER" ]; then
  ISSUE_REF="${ISSUE_OWNER}/${ISSUE_REPO}#${ISSUE_NUMBER}"
else
  ISSUE_REF="https://github.com/${ISSUE_OWNER}/${ISSUE_REPO}/issues/${ISSUE_NUMBER}"
fi

# Append link to body
NEW_BODY="${CURRENT_BODY}

${KEYWORD} ${ISSUE_REF}"

# Update PR using Python to handle JSON escaping properly
python3 << PYEOF
import json
import subprocess

payload = {
    "query": "mutation(\$prId: ID!, \$body: String!) { updatePullRequest(input: {pullRequestId: \$prId, body: \$body}) { pullRequest { url } } }",
    "variables": {
        "prId": "${PR_ID}",
        "body": $(echo "$NEW_BODY" | jq -Rs .)
    }
}

result = subprocess.run(
    ["gh", "api", "graphql", "--input", "-"],
    input=json.dumps(payload),
    capture_output=True,
    text=True
)

if result.returncode == 0:
    print("✅ Successfully linked PR to issue!")
    print(f"   PR: https://github.com/${PR_OWNER}/${PR_REPO}/pull/${PR_NUMBER}")
    print(f"   Issue: https://github.com/${ISSUE_OWNER}/${ISSUE_REPO}/issues/${ISSUE_NUMBER}")
else:
    print("❌ Error:", result.stderr)
    exit(1)
PYEOF```
